src/ws-reconnect.ts

```diff
@@ -114,12 +123,26 @@ export class WebSocketReconnector extends EventEmitter2 {
 
   private _open () {
     this._instance = new (this.WebSocket)(this._url)
-    this._instance.on('open', () => void this.emit('open'))
+    this._instance.on('open', () => {
+      this.emit('open')
+      this.heartbeat()
+    })
     this._instance.on('close', (code: number, reason: string) => this._reconnect(code))
     this._instance.on('error', (err: Error) => this._reconnect(err))
     this._instance.on('message', (data: WebSocket.Data) => void this.emit('message', data))
   }
 
+  heartbeat () {
+    if (this._heartbeatTimer) {
+      global.clearTimeout(this._heartbeatTimer)
+    }
+
+    if (this._instance && this._instance.readyState === this._instance.OPEN) {
+      this._instance.ping()
+      this._heartbeatTimer = setTimeout(() => this.heartbeat(), 5000)
```

Comment:
`5000` is pretty frequent just to keep a connection alive, isn't it? I'd think 30 seconds would be fine.

Reply:
Is there somewhere in the middle (e.g. 15-20s)? We experience connection drops every 30 seconds or so on the low end, maybe every 45 seconds on the high end, if the connection is idle.

Reply:
20s sounds ok. Are you using a google load balancer like ingress? By default they close websockets after 30 seconds (https://cloud.google.com/load-balancing/docs/https/#websocket_proxy_support), which heartbeats won't change.